server/CGameHandler.cpp

```diff
@@ -5800,7 +5799,7 @@ void CGameHandler::runBattle()
 			const CGHeroInstance * curOwner = battleGetOwnerHero(next);
 
 			if ((next->position < 0 || next->getCreature()->idNumber == CreatureID::BALLISTA)	//arrow turret or ballista
-				&& (!curOwner || curOwner->getSecSkillLevel(SecondarySkill::ARTILLERY) == 0)) //hero has no artillery
+				&& (!curOwner || getRandomGenerator().nextInt(99) >= curOwner->valOfBonuses(Bonus::MANUAL_CONTROL, CreatureID::BALLISTA)))
```

Comment:
What is this code doing and why?

Reply:
It (a) makes manual control configurable via the bonus system, rather than hacing it hard-coded to a particular secondary skill, and (b) allows a percentage-chance of control (rather than always or never).

Reply:
Ok. For now it's fine as long as default behavior stay 100% the same.
I suppose value will be 100 by default for heries with artillery skill, right?

Though in future it's would be great if you can separate underlying engine changes from changes that might alter gameplay if something go wrong.

Reply:
Yes, default for heroes with artillery will be 100, so no changes to default behaviour.